packages/firestore/src/util/misc.ts

```diff
@@ -74,6 +74,22 @@ export interface Equatable<T> {
   isEqual(other: T): boolean;
 }
 
+/** Compare strings in UTF-8 encoded byte order */
+export function compareUtf8Strings(left: string, right: string): number {
+  // Convert the string to UTF-8 encoded bytes
+  const encodedLeft = new TextEncoder().encode(left);
+  const encodedRight = new TextEncoder().encode(right);
```

Comment:
can you do
```js
return compareBlobs(encodedLeft, encodedRight);
```
?

Reply:
No, we can't. This is intentionally done in this way to avoid circular dependency, both having the `compareUtf8Strings` function in this file, and not importing 'compareBlobs` into it.

Reply:
I am open to suggestions tho.